### ISSUE TITLE
Overriding (*args: Any) and (**kwargs: Any) for args report error

### ISSUE THREAD

**Issue by fi6** [microsoft/pyright#1381]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
When overriding function with args using *args and **kwargs with type Any, pyright report error "reportIncompatibleMethodOverride"

**To Reproduce**
Steps to reproduce the behavior.
```python
Class A(ABC):
    @abstractmethod
    def execute(self, *args: Any, **kwargs: Any) -> Any:
        """
        docstring
        """
        raise NotImplementedError

Class B(A):
    async def execute(self, command_str: str, args: Sequence[str],
                      msg: TextMsg) -> Result or None:
        return await self.run_func(Session(
            self,
            command_str,
            args,
            msg,
        ))
```

Method "execute" overrides class "A" in an incompatible manner
  Parameter count mismatch; base method has 3, but override has 4
  Parameter 2 type mismatch: base parameter is type "Any", override parameter is type "str"
  Parameter 3 type mismatch: base parameter is type "Any", override parameter is type "Sequence[str]"

**Expected behavior**
Should not be an error.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
Given above

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

PyLance

**Additional context**

**Comment by erictraut** [comment]
The purpose of the reportIncompatibleMethodOverride is to ensure that the subclass is fulfilling the contract of the parent class when overriding a method. In your example, the two method signatures bear no relationship to another. Once is async, the other is not. One of them accepts any number of parameters of any type whereas the other accepts very specific parameters with specific types. If you were to pass an object of type `A` to an interface, what parameters would you expect it to call `execute` with if every (non-abstract) subclass can define different signatures?

I would consider the code in your example to be an incorrect and dangerous use of a method override, but if you want to continue to use this pattern, you can disable the `reportIncompatibleMethodOverride` diagnostic check for your entire project or within a single file (using `# pyright: reportIncompatibleMethodOverride=false`.

In any case, I consider the correct behavior of pyright to be correct in this case.

**Comment by fi6** [comment]
So let me 

> The purpose of the reportIncompatibleMethodOverride is to ensure that the subclass is fulfilling the contract of the parent class when overriding a method. In your example, the two method signatures bear no relationship to another. Once is async, the other is not. One of them accepts any number of parameters of any type whereas the other accepts very specific parameters with specific types. If you were to pass an object of type `A` to an interface, what parameters would you expect it to call `execute` with if every (non-abstract) subclass can define different signatures?
> 
> I would consider the code in your example to be an incorrect and dangerous use of a method override, but if you want to continue to use this pattern, you can disable the `reportIncompatibleMethodOverride` diagnostic check for your entire project or within a single file (using `# pyright: reportIncompatibleMethodOverride=false`.
> 
> In any case, I consider the correct behavior of pyright to be correct in this case.

So please let me change my example. In the actual case, whenever we use *args or **kwargs, pyright will treat this as a single argument, and I think this is not as designed for python.

in class A:
```py
    @abstractmethod
    async def execute(self, command_str: str, arg_list: Sequence[str],
                      msg: BaseMsg, *args: Any) -> Any:
        """
        docstring
        """
        raise NotImplementedError
```
then in class B:
```py
    async def execute(self, command_str: str, args: Sequence[str],
                      msg: TextMsg, other1: str, other2: str) -> Result or None:
        return await self.run_func(Session(
            self,
            command_str,
            args,
            msg,
        ))
```
will still reproduce the error: Parameter count mismatch; base method has 5, but override has 6

**Comment by erictraut** [comment]
This is the same problem. If you pass an object of type `B` to a function that expects an object of type `A`, how should it invoke the `execute` method in a way that it is guaranteed to work with `B` and any other subclass of `A`?

Consider the following example:

```python
async def my_func(obj: A):
    await obj.execute('cmd', [], TextMsg(), 1, 2, 3, 4, 5, 6) # This will crash

x = my_func(B())
```

This would type check without a problem, but it would generate a runtime error because the `execute` method defined in `B` does not accept that many arguments. This is the type of bug that the `reportIncompatibleMethodOverride` diagnostic check is designed to catch. If it were modified in the manner that you're suggesting, a whole class of bugs would go uncaught.

**Comment by fi6** [comment]
> This is the same problem. If you pass an object of type `B` to a function that expects an object of type `A`, how should it invoke the `execute` method in a way that it is guaranteed to work with `B` and any other subclass of `A`?
> 
> Consider the following example:
> 
> ```python
> async def my_func(obj: A):
>     await obj.execute('cmd', [], TextMsg(), 1, 2, 3, 4, 5, 6) # This will crash
> 
> x = my_func(B())
> ```
> 
> This would type check without a problem, but it would generate a runtime error because the `execute` method defined in `B` does not accept that many arguments. This is the type of bug that the `reportIncompatibleMethodOverride` diagnostic check is designed to catch. If it were modified in the manner that you're suggesting, a whole class of bugs would go uncaught.

Make sense. Thanks for you reply! 
I have tried to add *args to method of B and it works. That's really awesome to catch such potential bug with everything under consideration！
Also, I would suggest to make the diagnostic report more detailed and tell users that they should add *args to their override method as well, because I wrongly think that pyright treat *args as only one arg, but actually it is not.
